test/Microsoft.SqlTools.ServiceLayer.UnitTests/EditData/Common.cs

```diff
@@ -133,6 +91,29 @@ public static DbDataReader GetNewRowDataReader(DbColumn[] columns, bool includeI
             return new TestDbDataReader(new [] {testResultSet}, false);
         }
 
+        public static EditTableMetadata GetCustomEditTableMetadata(DbColumn[] columns)
+        {
+            // Create column metadata providers            
+            var columnMetas = columns.Select((c, i) => new EditColumnMetadata
```

Comment:
Minor: given you're iterating over columns twice, you could just create 2 lists and in a foreach loop assign to each one. Given the num columns is usually small, it's OK if you choose to ignore (just we should always be aware of these kind of optimizations)

Reply:
Just realized this is test code... ignore my previous comment

Reply:
You're still right, I'll make the change